unumpy/_multimethods.py

```diff
@@ -1154,3 +1156,23 @@ def _allclose_default(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
 @all_of_type(ndarray)
 def allclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
     return a, b
+
+
+def diag_default(v, k=0):
+    if ndim(v) == 1:
+        zeros = [0 for _ in range(len(v) * len(v))]
+        zeros[0 :: len(v) + 1] = v
+        return array(zeros).reshape((len(v), len(v)))
+    elif ndim(v) == 2:
+        if k < 0:
+            return array([v[i + abs(k), i] for i in range(N - abs(k))])
```

Comment:
And indexing here.

Reply:
Not sure I understand what you mean by indexing here. Could you elaborate a bit more?

Reply:
TBH, I'd remove the default for `diag` outright. It requires either large loops (which we don't like) or mutability, which a lot of libraries don't have.

Reply:
Okay, this answers the question I made seconds ago. Thank you!